src/App.tsx

```diff
@@ -15,47 +18,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export function getSortedGoods(
+  goods: string[],
+  state: State,
+) {
+  const { sortType, isReversed } = state;
+  const visibleGoods = [...goods].sort((goodA, goodB) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const resetOrder = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const visibleGoods = getSortedGoods(goodsFromServer,
+    { sortType, isReversed });
```

Comment:
well k, what you're passing to the function is not a state actually. Why is it called like that then?